splunk-otel-android/src/test/java/com/splunk/rum/DiskToZipkinExporterTest.java

```diff
@@ -0,0 +1,161 @@
+package com.splunk.rum;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import static java.util.Collections.singletonList;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import zipkin2.Call;
+import zipkin2.reporter.Sender;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DiskToZipkinExporterTest {
+
+    static final int BANDWIDTH_LIMIT = 20 * 1024;
+    static final Path spanFilesPath = Paths.get("/path/to/thing");
+    private final byte[] span1 = "span1".getBytes(StandardCharsets.UTF_8);
+    private final byte[] span2 = "span2".getBytes(StandardCharsets.UTF_8);
+    private final byte[] span3 = "span3".getBytes(StandardCharsets.UTF_8);
+    private final Path file1 = spanFilesPath.resolve("file1.spans");
+    private final Path file2 = spanFilesPath.resolve("file2.spans");
+    private final List<byte[]> file1Spans = singletonList(span1);
+    private final List<byte[]> file2Spans = Arrays.asList(span2, span3);
+    private final Path imposter = spanFilesPath.resolve("someImposterFile.dll");
+
+    @Mock
+    private ConnectionUtil connectionUtil;
+    @Mock
+    private FileUtils fileUtils;
+    @Mock
+    private CurrentNetwork currentNetwork;
+    @Mock
+    Sender sender;
+    @Mock
+    private BandwidthTracker bandwidthTracker;
+
+    @Before
+    public void setup() throws Exception{
+        when(connectionUtil.refreshNetworkStatus()).thenReturn(currentNetwork);
+        when(currentNetwork.isOnline()).thenReturn(true);
+        Stream<Path> files = Stream.of(file1, imposter, file2);
+        when(fileUtils.listFiles(spanFilesPath)).thenReturn(files);
+        when(fileUtils.isRegularFile(file1)).thenReturn(true);
+        when(fileUtils.isRegularFile(file2)).thenReturn(true);
+        when(fileUtils.isRegularFile(imposter)).thenReturn(true);
+        when(fileUtils.readFileCompletely(file1)).thenReturn(file1Spans);
+        when(fileUtils.readFileCompletely(file2)).thenReturn(file2Spans);
+    }
+
+    @Test
+    public void testHappyPathExport() throws Exception {
```

Comment:
Can you add some verifications that `bandwidthTracker.tick()` was called?

Reply:
Good call. 👍🏻